### PR TITLE
Fixup IMB to be more configurable

### PR DESCRIPTION
**Pull request by rfbgo**
- Make multiping pongs multi mode configurable
- Allow users to pass custom args
- Remove unneeded args from executables
- Fix small fom typo that reversed avg and max

### CODE REVIEW

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py [hunk 1]
```diff
@@ -41,41 +41,46 @@ class IntelMpiBenchmarks(SpackApplication):
...
 
-    # FIXME: how best to deal with the desirded behavior of <2*num_cores> -ppn <num_cores>
     executable('multi-pingpong',
                '-genv I_MPI_FABRICS=shm:tcp '
                'IMB-MPI1 Pingpong -msglog {msglog_min}:{msglog_max} '
-               '-iter {num_iteration} -multi 0 -map 2*{num_cores} -show_tail on',
+               '-iter {num_iterations} -multi {multi_val} -map {map_args} {additional_args}',
                use_mpi=True)
 
+    workload_variable('num_cores',
```
Comment (rohitramu) [inline var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py:53]: I don't see any references to `num_cores` in the `multi-pingpong` executable definition on line 47.  Where would this variable get used?

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py [hunk 2]
```diff
@@ -41,41 +41,46 @@ class IntelMpiBenchmarks(SpackApplication):
...
+              description='Number of cores',
+              workloads=['multi-pingpong'])
+
+    workload_variable('multi_val',
+              default='0',
+              description='Value to pass to the multi arg',
+              workloads=['multi-pingpong'])
+
+    workload_variable('map_args',
+              default='{num_cores}x2',
```
[the remark this answers is not in the document]
Comment (rfbgo) [reply var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py:64]: This is actually ramble / python magic, and will be expanded before bash etc (other wise yes, I agree)

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py (hunk 1 above)
Comment (rfbgo) [reply var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py:53]: There is a fairly subtle layer of indirection here. variables can refer to each other. This is consumed here:

```
 63     workload_variable('map_args',
 64               default='{num_cores}x2',
```

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py [hunk 3]
```diff
@@ -103,6 +108,10 @@ class IntelMpiBenchmarks(SpackApplication):
                       description='Max Message Size (power of 2)',
                       workloads=['pingpong', 'multi-pingpong', 'collective'])
 
+    workload_variable('additional_args', default='',
```
[the remark this answers is not in the document]
Comment (rfbgo) [reply var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py:111]: I think this is right. My intent was to allow user to pass arbitrary extra args to each of the three executables, eg: https://github.com/GoogleCloudPlatform/ramble/pull/316/files#diff-9b14a253a1554993b92a9c221cc9654f594eab35433fd4b14066c33377953b74

Do you see something here that concerns you?

[the remark this answers is not in the document]
Comment (rohitramu) [reply var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py:111]: Nope, that makes sense.  Thanks!

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py (hunk 1 above)
Comment (rohitramu) [reply var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py:53]: Got it, thanks!

**Review by rohitramu — APPROVED** [review]

**Review by douglasjacobsen — APPROVED** [review]

**Pull request merged**